src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/issuefinaldecision/IssueFinalDecisionAboutToSubmitHandler.java

```diff
@@ -87,6 +110,12 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         return preSubmitCallbackResponse;
     }
 
+    private final Predicate<SscsCaseData> eligibleForHearingsCancel = sscsCaseData -> scheduleListingEnabled
+            && List.of(State.HEARING, State.READY_TO_LIST).contains(Optional.of(sscsCaseData)
+                    .map(SscsCaseData::getState).orElse(State.UNKNOWN))
```

Comment:
Optional: split out these boolean clauses for each &&, assigning them meaningful boolean names. Concision can sometimes come at the cost of readability.

Reply:
I have refactored to make the subconditions more readable.

Reply:
LGTM